### PR TITLE
add reward protocol transaction to log

### PR DESCRIPTION
**Pull request by yutongp**

### CODE REVIEW

File: action/protocol/rewarding/fund.go [hunk 1]
```diff
@@ -14,9 +14,12 @@ import (
 	"github.com/pkg/errors"
 
 	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-core/action"
 	"github.com/iotexproject/iotex-core/action/protocol"
 	accountutil "github.com/iotexproject/iotex-core/action/protocol/account/util"
 	"github.com/iotexproject/iotex-core/action/protocol/rewarding/rewardingpb"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
```
Comment (dustinxie) [inline action/protocol/rewarding/fund.go:22]: this goes before /iotex-core

File: action/protocol/rewarding/fund.go [hunk 2]
```diff
@@ -59,31 +62,48 @@ func (p *Protocol) Deposit(
...
 	f.totalBalance = big.NewInt(0).Add(f.totalBalance, amount)
 	f.unclaimedBalance = big.NewInt(0).Add(f.unclaimedBalance, amount)
-	return p.putState(ctx, sm, fundKey, &f)
+	if err := p.putState(ctx, sm, fundKey, &f); err != nil {
+		return nil, err
+	}
+	return &action.Log{
+		Address: p.addr.String(),
+		Topics: action.Topics{
+			hash.BytesToHash256([]byte{byte(iotextypes.TransactionLogType_DEPOSIT_TO_REWARDING_FUND)}),
```
Comment (dustinxie) [inline action/protocol/rewarding/fund.go:96]: define this as DepositToRewardingTopic, see receipt.go

File: action/protocol/rewarding/reward.go [hunk 3]
```diff
@@ -278,15 +280,33 @@ func (p *Protocol) Claim(
...
 	}
-	return p.claimFromAccount(ctx, sm, actionCtx.Caller, amount)
+	if err := p.claimFromAccount(ctx, sm, actionCtx.Caller, amount); err != nil {
+		return nil, err
+	}
+
+	return &action.Log{
+		Address: p.addr.String(),
+		Topics: action.Topics{
+			hash.BytesToHash256([]byte{byte(iotextypes.TransactionLogType_CLAIM_FROM_REWARDING_FUND)}),
```
Comment (dustinxie) [inline action/protocol/rewarding/reward.go:299]: define this as ClaimFromRewardingTopic, see receipt.go

File: action/protocol/rewarding/protocol.go [hunk 4]
```diff
@@ -373,13 +375,20 @@ func (p *Protocol) createReceipt(
 	gasConsumed uint64,
 	logs ...*action.Log,
 ) *action.Receipt {
+	// remove possible nil log
+	nlogs := make([]*action.Log, 0, len(logs))
+	for _, l := range logs {
+		if l != nil {
+			nlogs = append(nlogs, l)
+		}
+	}
```
Comment (dustinxie) [inline action/protocol/rewarding/protocol.go:384]: better off to use []*action.Log as input, to avoid this hassle

like settleAction() in staking protocol

Comment (yutongp) [reply action/protocol/rewarding/protocol.go:384]: *action.Log in []*action.Log slice can be still nil

**Pull request merged**